### DISCUSSION TITLE
Can message handlers be private?

### DISCUSSION THREAD

**Discussion by dAnjou** [Textualize/textual#3391] [Q&A] [answered]
Hi,

I saw these two methods in `DirectoryTree`:
https://github.com/Textualize/textual/blob/4d1f057968fcef413e1ec4d4f210440a9b46db8a/src/textual/widgets/_directory_tree.py#L427
https://github.com/Textualize/textual/blob/4d1f057968fcef413e1ec4d4f210440a9b46db8a/src/textual/widgets/_directory_tree.py#L437

Both are not overrides of `Tree` methods, and they are not called anywhere. Which makes me think that message handlers can be private. But it's not documented, for example here:

https://textual.textualize.io/guide/events/#handler-naming
https://textual.textualize.io/widgets/tree/#messages

Is there any difference between prefixing handlers with `on_` or `_on_`? And can you elaborate whether others should consider using `_on_`?

Thanks 🙏

**Comment by willmcgugan** [comment, marked as answer]
They can indeed be private. This was to allow widgets in the code library to define handlers, without forcing subclasses to stick to the same signature.
